activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -481,7 +480,9 @@ def find_one(id)
       def find_some(ids)
         return find_some_ordered(ids) unless order_values.present?
 
-        result = where(primary_key => ids).to_a
+        relation = where(primary_key => ids)
+        relation = relation.select(table[primary_key]) unless select_values.empty?
+        result = relation.to_a
```

Comment:
Do we even need it here? `find_some` doesn't need the primary key, does it?

Reply:
It does not need them, but I tried to make it consistent, so both of them always select `id`s.
Having these ids, we can also slightly improve the error reporting here: https://github.com/rails/rails/blob/4d4f0b2abdee68693550886f4639f492ef14c570/activerecord/lib/active_record/relation/finder_methods.rb#L501
by passing not found ids.

Wdyt?